### PR TITLE
Core: Allow linking to kind/component ID

### PR DESCRIPTION
**Pull request by lonyele**
Issue: #7582 

## What I did
I added the default route when story exists but child url is not specified

## How to test

- Is this testable with Jest or Chromatic screenshots? Yes!(I added)
- Does this need a new example in the kitchen sink apps? No
- Does this need an update to the documentation? No

### CODE REVIEW

**Review by tmeasday — APPROVED** [review]
Seems reasonable. Should we deprecate the `--*` syntax?

File: lib/api/src/tests/stories.test.js [hunk 1]
```diff
@@ -207,6 +207,25 @@ describe('stories API', () => {
       expect(navigate).toHaveBeenCalledWith('/story/a--1');
     });
 
+    it(
+      'navigates to the first child story of the parent story if a ' +
```
Comment (shilman) [inline lib/api/src/tests/stories.test.js]: @lonyele what's up with this string concat?

File: lib/api/src/modules/stories.ts [hunk 2]
```diff
@@ -286,6 +286,11 @@ Did you create a path that uses the separator char accidentally, such as 'Vue <d
       } else if (viewMode && firstLeaf) {
         navigate(`/${viewMode}/${firstLeaf.id}`);
       }
+    } else if (storiesHash[storyId] && storiesHash[storyId].children) {
```
Comment (shilman) [inline lib/api/src/modules/stories.ts]: @lonyele will this navigate multiple times if `storyId` is not immediately above the leaf?

File: lib/api/src/tests/stories.test.js (hunk 1 above)
Comment (lonyele) [reply lib/api/src/tests/stories.test.js]: It's not a prettier error(80 character) but this string became too long and I ended up concatenating like this. Is there  a better way? maybe template literal?

**Comment by lonyele** [comment]
@shilman  just a gentle ping...

**Comment by shilman** [comment]
Thanks @lonyele! Will merge this into 5.3 after 5.2 is released later this week. 👍

File: lib/api/src/modules/stories.ts (hunk 2 above)
Comment (shilman) [reply lib/api/src/modules/stories.ts]: `storiesHash[storyId].children` is not necessarily a leaf, is it? It's just a child node ID. What happens in the case that the child is not a leaf?

File: lib/api/src/modules/stories.ts [hunk 3]
```diff
@@ -286,11 +297,14 @@ Did you create a path that uses the separator char accidentally, such as 'Vue <d
...
       }
-    } else if (storiesHash[storyId] && storiesHash[storyId].children) {
-      // When story exists but specific child is not specified, it routes
-      // to the first child of the story.
-      const firstLeafId = storiesHash[storyId].children[0];
-      navigate(`/${viewMode}/${firstLeafId}`);
+    } else if (storiesHash[storyId]) {
+      // When story exists but if it is not the leaf story, it finds the proper
+      // leaf story from any depth.
+      const firstLeafStoryId = storiesHash[storyId].isLeaf
```
Comment (shilman) [inline lib/api/src/modules/stories.ts]:
```js
const firstLeafStoryId = findLeafStoryId(storiesHash, storyId);
```

Comment (shilman) [reply lib/api/src/modules/stories.ts]: @lonyele

Comment (lonyele) [reply lib/api/src/modules/stories.ts]: @shilman oh wait, I'm writing a test now

**Comment by lonyele** [comment]
@shilman It's much better now. It will navigate to the leaf story from any depth. Consider the cases through this screenshot. 
SB-default-any-depth

URL from the screenshot is `/?path=/story/basics-form-select--validations`

- if given url is `/?path=/story/basics-form-select` then
  navigates to `/?path=/story/basics-form-select--sizes` 
- if given url is `/?path=/story/basics-form` then
  navigates to `/?path=/story/basics-form-button--sizes` 
- if given url is `/?path=/story/basics` then
  navigates to `/?path=/story/basics-actionbar--single-item` 


btw, I added 2 test cases for different depth. If test case naming or too many cases, please let me know how to fix it(I think it is little bit weird now)

File: lib/api/src/tests/stories.test.js (hunk 1 above)
Comment (lonyele) [reply lib/api/src/tests/stories.test.js]: I changed it to one line!

**Pull request merged**